taprpc/taprpc_utils.go

```diff
@@ -0,0 +1,42 @@
+package taprpc
+
+import "google.golang.org/protobuf/encoding/protojson"
+
+var (
+	// ProtoJSONMarshalOpts is a struct that holds the default marshal
+	// options for marshaling protobuf messages into JSON in a
+	// human-readable way. This should only be used in the CLI and in
+	// integration tests.
+	ProtoJSONMarshalOpts = &protojson.MarshalOptions{
+		EmitUnpopulated: true,
+		UseProtoNames:   true,
+		Indent:          "    ",
+		UseHexForBytes:  true,
+	}
+
+	// ProtoJSONUnmarshalOpts is a struct that holds the default unmarshal
+	// options for un-marshaling lncli JSON into protobuf messages. This
+	// should only be used in the CLI and in integration tests.
+	ProtoJSONUnmarshalOpts = &protojson.UnmarshalOptions{
+		AllowPartial:   false,
+		UseHexForBytes: true,
+	}
+
+	// RESTJsonMarshalOpts is a struct that holds the default marshal
+	// options for marshaling protobuf messages into REST JSON in a
+	// human-readable way. This should be used when interacting with the
+	// REST proxy only.
+	RESTJsonMarshalOpts = &protojson.MarshalOptions{
+		EmitUnpopulated: true,
+		UseProtoNames:   true,
+		UseHexForBytes:  true,
+	}
+
+	// RESTJsonUnmarshalOpts is a struct that holds the default unmarshal
+	// options for un-marshaling REST JSON into protobuf messages. This
+	// should be used when interacting with the REST proxy only.
+	RESTJsonUnmarshalOpts = &protojson.UnmarshalOptions{
+		AllowPartial:   false,
+		UseHexForBytes: true,
```

Comment:
q: I guess it's to late to change this in [LND](https://github.com/lightningnetwork/lnd/blob/ac3e7be9a063bdb559855f3b5af15e0323878450/lnrpc/rpc_utils.go#L44) too because other tools are using already base64 encoding?

Reply:
I thought about adding an option to give the user the choice. But the problem is that yes, most of the apps using the REST interface now rely on the raw bytes being base64 encoded. So it shouldn't be the user's choice but rather the developer's. This gives me an idea though: Maybe there is a way to specify the desired encoding as a HTTP header field? For example: `Accept: application/json+hex` instead of just `Accept: application/json`...